File: backend/src/main/java/com/lsoft/aguapotable/controller/ReporteController.java
package com.lsoft.aguapotable.controller;

import com.lsoft.aguapotable.service.ReporteService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.servlet.http.HttpServletRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.math.BigDecimal;
import java.util.Map;

@RestController
@RequestMapping("/api/v1/reportes")
@RequiredArgsConstructor
@Tag(name = "Reportes", description = "Dashboard y reportes operativos")
@SecurityRequirement(name = "bearerAuth")
public class ReporteController {

    private final ReporteService reporteService;

    @GetMapping("/dashboard")
    @Operation(summary = "Datos del dashboard: KPIs principales de la comunidad")
    public ResponseEntity<Map<String, Object>> getDashboard(HttpServletRequest request) {
        Long comunidadId = (Long) request.getAttribute("comunidadId");
        return ResponseEntity.ok(reporteService.getDashboard(comunidadId));
    }

    @GetMapping("/morosidad")
    @Operation(summary = "Reporte de cuentas morosas")
    public ResponseEntity<Map<String, Object>> getMorosidad(HttpServletRequest request) {
        Long comunidadId = (Long) request.getAttribute("comunidadId");
        return ResponseEntity.ok(reporteService.getReporteMorosidad(comunidadId));
    }

    @GetMapping("/recaudacion-mensual")
    @Operation(summary = "Recaudación de los últimos 12 meses (para gráfica)")
    public ResponseEntity<Map<String, BigDecimal>> getRecaudacionMensual(HttpServletRequest request) {
        Long comunidadId = (Long) request.getAttribute("comunidadId");
        return ResponseEntity.ok(reporteService.getRecaudacionMensual(comunidadId));
    }
}

File: backend/src/reporte/reporte.controller.ts
import { Controller, Get, UseGuards } from '@nestjs/common';
import { AuthGuard } from '@nestjs/passport';
import { ApiTags, ApiOperation, ApiBearerAuth } from '@nestjs/swagger';
import { ReporteService } from './reporte.service';
import { ComunidadId } from '../common/decorators';
import { RolesGuard } from '../auth/guards/roles.guard';

@ApiTags('Reportes')
@ApiBearerAuth()
@Controller('api/v1/reportes')
@UseGuards(AuthGuard('jwt'), RolesGuard)
export class ReporteController {
  constructor(private readonly reporteService: ReporteService) {}

  @Get('dashboard')
  @ApiOperation({ summary: 'Datos del dashboard: KPIs principales' })
  async getDashboard(@ComunidadId() comunidadId: number) {
    return this.reporteService.getDashboard(comunidadId);
  }

  @Get('morosidad')
  @ApiOperation({ summary: 'Reporte de cuentas morosas' })
  async getMorosidad(@ComunidadId() comunidadId: number) {
    return this.reporteService.getReporteMorosidad(comunidadId);
  }

  @Get('recaudacion-mensual')
  @ApiOperation({ summary: 'Recaudación de los últimos 12 meses' })
  async getRecaudacionMensual(@ComunidadId() comunidadId: number) {
    return this.reporteService.getRecaudacionMensual(comunidadId);
  }
}
